moira/cache.py

```diff
@@ -1,9 +1,33 @@
 from functools import wraps
 
 from twisted.internet import reactor, defer
+from collections import OrderedDict
 
-CACHE = {}
+#CACHE = {}
+class CacheClass(OrderedDict):
+    """
+    Autocleaning dict.
+    """
+    
+    max_live_time = 2*60**2 # 2 hours
+    
+    def __setitem__(self, key, value, *args, **kwargs):
+        self._cleanup()
```

Comment:
This will be called always twice now. Look at line 49, 51.
Let's make public cleanup method and call it directly once.